scripts/twister

```diff
@@ -206,9 +206,23 @@ sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/pylib/build_helpers"))
 from twisterlib.environment import add_parse_arguments, parse_arguments
 from twisterlib.twister_main import main
 
+
+def python_version_guard():
+    min_ver = (3, 10)
+    if sys.version_info < min_ver:
+        raise EnvironmentError(
+          "Obsolete Python version!"
+          f" Currently, Twister requires at least Python {'.'.join(min_ver)}."
+          f" You are trying to use Python {sys.version}."
+          " Install a newer Python version and retry."
+        )
+
+
 if __name__ == "__main__":
     ret = 0
     try:
+        python_version_guard()
```

Comment:
can this check just print the message and exit gracefully instead of this confusing traceback ?,
why it needs to raise exception ?
```
Traceback (most recent call last):                                                                                           
  File "./scripts/twister", line 222, in <module>                                                                            
    python_version_guard()                                                                                                   
  File "./scripts/twister", line 212, in python_version_guard                                                                
    raise EnvironmentError(                                                                                                  
OSError: Obsolete Python version! Currently, Twister requires at least Python (3, 10). You are trying to use Python 3.8.10 (default, Jul 29 2024, 17:02:10)                                                                                               
[GCC 9.4.0]. Install a newer Python version and retry.                                                                       
```